components/Table/src/TableColGroup.tsx

```diff
@@ -0,0 +1,17 @@
+import React, { ColgroupHTMLAttributes } from 'react';
+import clsx from 'clsx';
+
+export type TableColGroupProps = ColgroupHTMLAttributes<HTMLTableColElement>;
+
+/**
+ * TableColGroup component based on the <colgroup> element.
+ * Defines a group of columns within a table.
+ */
+export const TableColGroup: React.FC<TableColGroupProps> = ({ className, ...props }: TableColGroupProps) => {
+  const rootClassNames = clsx('denhaag-table__colgroup', className);
```

Comment:
Inconsistent use of casing wrt the classname here.
Usually the places where we case a letter in a component name would lead to a `-` in the classname.